yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -62,50 +62,42 @@ private[yarn] class YarnAllocator(
     amClient: AMRMClient[ContainerRequest],
     appAttemptId: ApplicationAttemptId,
     args: ApplicationMasterArguments,
-    preferredNodes: collection.Map[String, collection.Set[SplitInfo]],
```

Comment:
This is discussed in the proposal - preferred locality on YARN is currently broken for a few other reasons.  The goal here is make the code as simple as possible, and we can later add it back in in a way that makes sense for SPARK-4352.